### ISSUE TITLE
enforce deterministic GPU algorithms and uniform random seed assignment

### ISSUE THREAD

**Issue by cameronraysmith** [pinellolab/pyrovelocity#478]
- [v2.6 torch reproducibility](https://pytorch.org/docs/2.6/notes/randomness.html#reproducibility)
- [v2.6 torch.use_deterministic_algorithms](https://pytorch.org/docs/2.6/generated/torch.use_deterministic_algorithms.html)
- [v2.2 torch.use_deterministic_algorithms](https://pytorch.org/docs/2.2/generated/torch.use_deterministic_algorithms.html)
- [v1.8 torch.use_deterministic_algorithms](https://pytorch.org/docs/1.8.0/generated/torch.use_deterministic_algorithms.html?highlight=deterministic#torch.use_deterministic_algorithms) 
- [CUDA cublas results reproducibility](https://docs.nvidia.com/cuda/cublas/index.html#results-reproducibility)

```python
import os

os.environ["CUBLAS_WORKSPACE_CONFIG"] = ":4096:8"

import random

import numpy as np
import torch

torch.use_deterministic_algorithms(
    mode=True,
    warn_only=True,
)

SEED = 1

torch.manual_seed(SEED)
random.seed(SEED)
np.random.seed(seed=SEED)
```
